### PR TITLE
dbeaver/pro#1641 Events handling

### PR DESCRIPTION
**Pull request by E1izabeth**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/EclipseWorkspaceImpl.java [hunk 1]
```diff
@@ -59,10 +60,22 @@ public EclipseWorkspaceImpl(DBPPlatform platform, IWorkspace eclipseWorkspace) {
...
     public final void initializeProjects() {
-        initializeWorkspaceSession();
-
-        loadWorkspaceProjects();
-
+        try {
+            initializeWorkspaceSession();
+    
+            loadWorkspaceProjects();
+        } catch (DBException e) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.registry/src/org/jkiss/dbeaver/registry/EclipseWorkspaceImpl.java]: This code can be executed only for auth error

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPProject.java [hunk 2]
```diff
@@ -108,7 +109,10 @@ public interface DBPProject extends DBPObject, SMAuthSpace, DBAPermissionRealm {
 
     @NotNull
     DBPDataSourceRegistry getDataSourceRegistry();
-
+    
+    @NotNull
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPProject.java]: Let's use another approach and don't touch DBPProject

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPWorkspaceDesktop.java [hunk 3]
```diff
@@ -44,7 +44,9 @@ public interface DBPWorkspaceDesktop extends DBPWorkspaceEclipse {
      * Reloads workspace contents. Creates missing projects, removes unexistent projects
      */
     void refreshWorkspaceContents(DBRProgressMonitor monitor) throws DBException;
-
+    
+    void reloadWorkspaceProjects(DBRProgressMonitor monitor) throws DBException;
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/app/DBPWorkspaceDesktop.java]: Let's move it to TE

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/preferences/SimplePreferenceStore.java [hunk 4]
```diff
@@ -35,14 +36,31 @@ public abstract class SimplePreferenceStore extends AbstractPreferenceStore {
     private Map<String, String> defaultProperties;
     private boolean dirty = false;
 
-    public SimplePreferenceStore()
-    {
+    public interface Mutable {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/preferences/SimplePreferenceStore.java]: Maybe we should just show notification about preferences change?

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/preferences/SimplePreferenceStore.java (hunk 4 above)
Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/preferences/SimplePreferenceStore.java]: There is enough synchronization for SQL Templates without additional synchronization in SimplePreferenceStore. We can handle cases with other settings when we will work on it.

**Pull request merged**